pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/RuntimeASTTransformer.groovy

```diff
@@ -1238,10 +1249,39 @@ class RuntimeASTTransformer {
             // These variable handles are declared in functions, but will still be bound to the script context
             // Doing this here ensures the variables make the trip across to run time - if they don't make it, neither did this pipeline
             pipelineBlock.addStatement(stmt(callX(closureX(currentBlock), 'call')))
-            pipelineElementHandles.clear()
         }
 
-        /**
+        @NonNull
+        private List<Statement> prepareClosureScopedHandles(@NonNull BlockStatement pipelineBlock) {
+            ArrayList<Statement> result = new ArrayList<Statement>()
+            if (SCRIPT_SPLITTING_TRANSFORMATION) {
+                ArrayList<DeclarationExpression> declarations = new ArrayList<DeclarationExpression>()
+                moduleNode.statementBlock.statements.each { item ->
+                    if (((ExpressionStatement) item).expression instanceof DeclarationExpression) {
```

Comment:
The error mentioned by @eplodn in https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/405#issuecomment-771677208 is because this is an unchecked cast, so anyone using `SCRIPT_SPLITTING_TRANSFORMATION` is probably hitting similar issues in the latest release. I will file a PR in a minute.